std/math.d

```diff
@@ -4616,6 +4616,14 @@ real hypot(real x, real y) @safe pure nothrow @nogc
 
     static assert(2*(SQRTMAX/2)*(SQRTMAX/2) <= real.max);
 
+    static if (floatTraits!(real).realFormat == RealFormat.ieeeQuadruple)
+    {
+        // This static assert overflows when cross-compiling from lower-precision
+        // floating-point reals at compile-time to higher precision at runtime.
```

Comment:
As noted, this isn't an issue just for Quadruples, but also hits when say the compile-time real is 64-bit and runtime is 80-bit, as I just tested with ldc on Android/ARM cross-compiling to linux/x64 (a Win/MSVC target works however, as it also uses 64-bit reals).

I'm unsure how to check the compile-time real's precision however, so I just settled for disabling this test whenever compiling to a Quad runtime real. Anyone know how to get the compile-time real's precision?

Reply:
@ibuclaw or @jpf91, any input?

Reply:
As noted in the forum, there's no way of detecting the host's compile-time precision. Compiled code shouldn't have to know, and an assert makes perfect sense to guarantee the precision is sufficiently high. It clearly isn't with current LDC in all cross-compilation scenarios, as we don't have a software `real_t` yet.

So maybe weakening the `static assert` to a runtime `enum <actual-ct-value> = ...; assert(...)` would be a solution for this problem, potentially restricted to LDC only.

Reply:
Doesn't affect gdc, as we have used soft float since the start.

Reply:
These static asserts assume IEEE though. Non-IEEE reals could (and infact do in the case of ibm real) throw a compile time assert.

Reply:
Alright, since this doesn't affect dmd, because it only supports Intel, or gdc, because it uses a wider-precision soft-float at compile-time, I'll keep this in ldc: removed here.